system-tests/test.py

```diff
@@ -179,16 +179,33 @@ def __init__(self,
                  fc_flags: str = None,
                  ld_flags: str = None):
         args: List[str] = []
+
         if fpp_flags:
-            args.append('--fpp-flags=' + fpp_flags)
-        if fc_flags:
-            args.append('--fc-flags=' + fc_flags)
-        if ld_flags:
-            args.append('--ld-flags=' + ld_flags)
+            # different config file name for fpp flag test
+            conf_file = test_directory/('stay_config.ini')
+        else:
+            conf_file = test_directory/'config.ini'
 
-        args.extend(['--exec-name', 'fab_test'])
-        args.append(target)
         args.append(str(test_directory))
+        args.append(str(conf_file))
+
+        configfile = open(conf_file, 'wt')
```

Comment:
I would recommend using the `with open(conf_file, 'wt') as configfile:` syntax here as it automatically handles a lot of fault conditions. You might also consider a """block string""" to save some typing.